department-of-veterans-affairs/LEAF
# Leaf 4301 - User Experience (Form Editor v2, Report Builder)

This makes Form Editor v2 the default option.

Other changes include:
- Fix "advanced options are present" indicator
- Change indicator for "has conditional logic" from a grey background to an icon
- Updates verbiage for "Collaborators" -- now called "Customize Write Access", and moved the button
- CSS design changes
- Updates Report Builder so that user provided titles are mirrored in the HTML title tag, which makes copy/pasting links a little more pleasant
- Remove old IE polyfill in the Report Builder-
- Adds missing "html/htmlPrint advanced options" for section headings
- Users are prompted to enter a short label if their form field contains rich text

### Prerequisites
- [x] Resolve blocker related to HTML entity issue @aerinkayne 

### Potential Impact
- Form Editor v2 significantly changes the user experience and will require notification to users. A "Back to old Form Editor" link is provided in the event v2 is missing functionality
- ~~The CSS changes fix alignment, but might break alignment in other areas? @aerinkayne~~
- Report Builder won't work in IE 🎉

### Testing
- [ ] The "advanced options are present" indicator shows up accurately
- [ ] The "conditional logic is present" indicator shows up accurately
- [ ] After labeling a report in the Report Builder, copy/pasting the link into Outlook or Teams shows the label instead of the generic "Report Builder..."
- [ ] HTML entities do not show up when editing headers
  - Steps to reproduce:
  1. Open a form in form_vue
  2. Click "Add Section"
  3. Click "Advanced Formatting"
  4. Type "test", and change it to "header 1"
  5. Click "Save". You should not see "\<h1>test\</h1>"
